src/App.tsx

```diff
@@ -1,23 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsOnPage, setItemsOnPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const itemsStartRange = (
+    (currentPage * itemsOnPage) - (itemsOnPage - 1)
+  );
+
+  const itemsEndRange = Math.min(
+    currentPage * itemsOnPage, items.length,
+  );
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${itemsStartRange} - ${itemsEndRange} of ${items.length})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            value={itemsOnPage}
+            onChange={
+              event => {
+                setItemsOnPage(+event.target.value);
+                setCurrentPage(1);
+              }
+            }
```

Comment:
better to create separate function
```suggestion
            onChange={onSelectChange}
```